pkg/rpc/client_util.go

```diff
@@ -164,30 +88,36 @@ func (w *wrappedClientStream) SendMsg(m interface{}) error {
 }
 
 func streamClientInterceptor(ctx context.Context, desc *grpc.StreamDesc, cc *grpc.ClientConn, method string, streamer grpc.Streamer, opts ...grpc.CallOption) (grpc.ClientStream, error) {
-	s, err := streamer(ctx, desc, cc, method, opts...)
+	s, err := ExecuteWithRetry(func() (interface{}, error) {
+		s, err := streamer(ctx, desc, cc, method, opts...)
+		if err != nil {
+			err = convertClientError(err)
+			logger.GrpcLogger.Errorf("create client stream error: %v for method: %s target: %s connState: %s", err, method, cc.Target(), cc.GetState().String())
+			return nil, err
+		}
+		return s, nil
+	}, 0.2, 2.0, 3, nil)
 	if err != nil {
-		err = convertClientError(err)
-		logger.GrpcLogger.Errorf("create client stream error: %v for method: %s target: %s connState: %s", err, method, cc.Target(), cc.GetState().String())
 		return nil, err
 	}
-
 	return &wrappedClientStream{
-		ClientStream: s,
+		ClientStream: s.(grpc.ClientStream),
 		method:       method,
 		cc:           cc,
 	}, nil
 }
 
 func unaryClientInterceptor(ctx context.Context, method string, req, reply interface{}, cc *grpc.ClientConn, invoker grpc.UnaryInvoker, opts ...grpc.CallOption) error {
-	messageSent.Event(ctx, 1, req)
-	err := invoker(ctx, method, req, reply, cc, opts...)
-
-	messageReceived.Event(ctx, 1, reply)
-	if err != nil {
-		err = convertClientError(err)
-		logger.GrpcLogger.Errorf("do unary client error: %v for method: %s target: %s connState: %s", err, method, cc.Target(), cc.GetState().String())
-	}
-
+	_, err := ExecuteWithRetry(func() (interface{}, error) {
```

Comment:
Use [go-grpc-middleware](https://github.com/grpc-ecosystem/go-grpc-middleware)

Reply:
go-grpc-middleware/retry does not support bidirectional stream, but the ReportPieceResult mothod of scheduler is designed as a bidirectional stream.